File: android/app/src/main/java/edu/gatech/cs2340/rajsjarfiles/spacetrader/entity/universe/Universe.java
package edu.gatech.cs2340.rajsjarfiles.spacetrader.entity.universe;

import java.util.Collection;
import java.util.HashSet;
import java.util.Random;

/**
 * Represents the entire universe of the game.
 */
public class Universe {
    private static final int MAX_SOLAR_SYSTEMS = 20;

    //using an array because the size won't change
    private final SolarSystem[] solarSystem;

    /**
     * Constructor to create the universe.
     */
    public Universe() {
        solarSystem = new SolarSystem[MAX_SOLAR_SYSTEMS];
        createUniverse();
    }

    /**
     * Method that creates a universe with multiple solar systems with unique
     * names and coordinates.
     */
    private void createUniverse() {
        for (int i = 0; i < solarSystem.length; i++) {
            Collection<String> nameSet = new HashSet<>();
            Collection<Coordinate> coordSet = new HashSet<>();
            String name = PlanetNames.generateName();
            Coordinate coordinate = new Coordinate();
            while (!nameSet.add(name)) {
                name = PlanetNames.generateName();
            }
            while (!coordSet.add(coordinate)) {
                coordinate = new Coordinate();
            }
            solarSystem[i] = new SolarSystem(name, coordinate);
        }
    }

    /**
     * @return a random solar system in the universe
     */
    public SolarSystem getRandomSolarSystem() {
        Random rand = new Random();
        return solarSystem[rand.nextInt(solarSystem.length)];
    }

    @Override
    public String toString() {
        String ret = "The universe:\n";
        for (SolarSystem aSolarSystem : solarSystem) {
            ret += "- " + aSolarSystem.toString() + "\n";
        }
        return ret;
    }
}

File: ios/SpaceTrader/SpaceTrader/src/entity/universe/Universe.swift
//
//  .swift
//  SpaceTrader
//
//  Created by Thomas Suarez on 4/20/19.
//  Copyright © 2019 RajsJarFiles. All rights reserved.
//

import Foundation

class Universe : CustomStringConvertible {
    
    static var MAX_SOLAR_SYSTEMS : Int = 20
    
    var solarSystem : [SolarSystem]
    
    init() {
        self.solarSystem = [SolarSystem]()
        self.createUniverse()
    }
    
    var description: String {
        get {
            return "The universe!"
        }
    }
    
    func createUniverse() {
        for i in 0..<self.solarSystem.count {
            var nameSet : [String] = [String]()
            var coordSet : [Coordinate] = [Coordinate]()
            let name : String = PlanetNames.generateName()
            let coordinate : Coordinate = Coordinate(x: 0, y: 0)
            solarSystem[i] = SolarSystem(name: name, coordinate: coordinate)
        }
    }
    
    func getRandomSolarSystem() -> SolarSystem {
        let r = Int.random(in: 0..<self.solarSystem.count)
        return self.solarSystem[r]
    }
    
}
